Ejercicio1 - Utilizar operadores aritméticos
Operadores aritméticos en Python
Python proporciona operadores aritméticos comunes para que puedas realizar operaciones matemáticas en tu código. Estos incluyen las cuatro operaciones principales de suma, resta, multiplicación y división.

Exploremos cómo podemos crear un programa que pueda calcular la distancia entre dos planetas. Comenzaremos usando dos distancias de planetas: Tierra (149.597.870 km) y Júpiter (778.547.200 km).

In [1]:
dist_tierra =149597870
dist_júpiter =778547200

-628949330


Realizar la operación
Con los valores obtenidos, es el momento de añadir el código para realizar la operación. Restarás el primer planeta del segundo para determinar la distancia en kilómetros. A continuación, puedes convertir la distancia del kilómetro en millas multiplicándola por 0.621.

# Calcular la distancia entre planetas
Prueba tu proyecto
Con el código creado, ejecuta el notebook para obtener el resultado. Deberías recibir lo siguiente:

628949330
390577534
Recuerda: Lo importante es intentarlo, puedes auxiliarte de internet, libros, etc. El objetivo de una kata es la comprobación de la comprensión de la lección y su aplicación inmediata a la resolución de problemas. Go for it!



In [6]:
dist1=dist_júpiter-dist_tierra
km_mill=dist1*0.621
print("El total de distancia en millas de júpiter a la tierra es de: ", km_mill," millas")

El total de distancia en millas de júpiter a la tierra es de:  390577533.93  millas


Ejercicio 2: convierte cadenas en números y usa valores absolutos
Crear una aplicación para trabajar con números y entrada de usuario
Con frecuencia, deberás convertir los valores de cadena en números para realizar correctamente diferentes operaciones o determinar el valor absoluto de un número.

Para crear nuestra aplicación, queremos leer la distancia del sol para dos planetas, y luego mostrar la distancia entre los planetas. Haremos esto usando input para leer los valores, int para convertir a entero y luego abs para convertir el resultado en su valor absoluto.

Lee los valores
Usando input, agrega el código para leer la distancia del sol para cada planeta, considerando 2 planetas.

# Almacenar las entradas del usuario
#Pista: variable = input("¿Cuál es tu nombre?")

In [16]:
plan1=input("Ingrese la distancia del sol al primer planeta")
plan2=input("Ingrese la distancia del sol al segundo planeta")

Convertir a número
Debido a que input devuelve valores de cadena, necesitamos convertirlos en números. Para nuestro ejemplo, usaremos int

# Convierte las cadenas de ambos planetas a números enteros

In [17]:
planeta1km= int (plan1)
planeta2km=int (plan2)

Realizar el cálculo y convertir a valor absoluto
Con los valores almacenados como números, ahora puedes agregar el código para realizar el cálculo, restando el primer planeta del segundo. Debido a que el segundo planeta podría ser un número mayor, usarás abs para convertirlo a un valor absoluto. También agregarás el código para mostrar el resultado en millas multiplicando la distancia del kilómetro por 0.621

In [18]:
dist=planeta2km-planeta1km
print(dist)

dist_m= abs(dist* 0.621)
print(dist_m)
print("la distancia final entre los planetas es de :", dist_m, "millas")

50300000
31236300.0
la distancia final entre los planetas es de : 31236300.0 millas


Prueba tu aplicación
Para probar el proyecto, ejecuta tu notebook. En la parte superior de vscode surgirá un cuadro de diálogo para que proporciones las distancias. Puede utilizar los datos de la tabla siguiente: